x/datapool/keeper/pool_test.go

```diff
@@ -138,3 +145,308 @@ func (suite *poolTestSuite) TestGetPool() {
+		DataHash:      dataHash,
+		DataValidator: dataVal1.String(),
+		Requester:     requesterAddr.String(),
+	}
+
+	bz, err := suite.Cdc.Marshaler.MarshalBinaryBare(&unsignedCert)
+	suite.Require().NoError(err)
+
+	sign, err := dataValPrivKey.Sign(bz)
+	suite.Require().NoError(err)
+
+	cert := types.DataValidationCertificate{
+		UnsignedCert: &unsignedCert,
+		Signature:    sign,
+	}
+
+	suite.DataPoolKeeper.SetDataValidationCertificate(suite.Ctx, cert)
+
+	getCert, err := suite.DataPoolKeeper.GetDataValidationCertificate(suite.Ctx, poolID, round, dataHash)
+	suite.Require().NoError(err)
+
+	suite.Require().Equal(cert.UnsignedCert.PoolId, getCert.UnsignedCert.PoolId)
+	suite.Require().Equal(cert.UnsignedCert.Round, getCert.UnsignedCert.Round)
+	suite.Require().Equal(cert.UnsignedCert.DataHash, getCert.UnsignedCert.DataHash)
+	suite.Require().Equal(cert.UnsignedCert.DataValidator, getCert.UnsignedCert.DataValidator)
+	suite.Require().Equal(cert.UnsignedCert.PoolId, getCert.UnsignedCert.PoolId)
+	suite.Require().Equal(cert.UnsignedCert.Requester, getCert.UnsignedCert.Requester)
+	suite.Require().Equal(cert.Signature, getCert.Signature)
+}
+
+func (suite *poolTestSuite) TestSellData() {
+	poolID := uint64(1)
+	round := uint64(1)
+	dataHash := []byte("dataHash")
+
+	validatorAccount := suite.AccountKeeper.NewAccountWithAddress(suite.Ctx, dataVal1)
+	err := validatorAccount.SetPubKey(dataValPubKey)
+	suite.Require().NoError(err)
+	suite.AccountKeeper.SetAccount(suite.Ctx, validatorAccount)
+
+	pool := makeTestDataPool(poolID)
+	suite.DataPoolKeeper.SetPool(suite.Ctx, pool)
+
+	cert, err := makeTestDataCertificate(suite.Cdc.Marshaler, poolID, round, dataHash)
+	suite.Require().NoError(err)
+
+	shareToken, err := suite.DataPoolKeeper.SellData(suite.Ctx, requesterAddr, *cert)
+	suite.Require().NoError(err)
+	suite.Require().Equal("DP/1", shareToken.Denom)
+	suite.Require().Equal(sdk.NewInt(1), shareToken.Amount)
+
+	// Check the current pool status
+	getPool, err := suite.DataPoolKeeper.GetPool(suite.Ctx, poolID)
+	suite.Require().NoError(err)
+	suite.Equal(uint64(1), getPool.CurNumData)
+	suite.Equal(types.PENDING, getPool.Status)
+}
+
+func (suite *poolTestSuite) TestSellData_change_status_activity() {
+	poolID := uint64(1)
+	round := uint64(1)
+	dataHash := []byte("dataHash")
+
+	validatorAccount := suite.AccountKeeper.NewAccountWithAddress(suite.Ctx, dataVal1)
+	err := validatorAccount.SetPubKey(dataValPubKey)
+	suite.Require().NoError(err)
+	suite.AccountKeeper.SetAccount(suite.Ctx, validatorAccount)
+
+	pool := makeTestDataPool(poolID)
+	// Modify the target data of the pool
+	pool.PoolParams.TargetNumData = 1
+	suite.DataPoolKeeper.SetPool(suite.Ctx, pool)
+
+	cert, err := makeTestDataCertificate(suite.Cdc.Marshaler, poolID, round, dataHash)
+	suite.Require().NoError(err)
+
+	shareToken, err := suite.DataPoolKeeper.SellData(suite.Ctx, requesterAddr, *cert)
+	suite.Require().NoError(err)
+	suite.Require().Equal("DP/1", shareToken.Denom)
```

Comment:
Our initial denom is `DP-{poolID}`
However, when verifying in Coin, the following regular expression is used.
`[a-zA-Z][a-zA-Z0-9/]{2,127}`
So I changed `-` to `/`